lib/beaker/dsl/install_utils/windows_utils.rb

```diff
@@ -143,7 +143,9 @@ def install_msi_on(hosts, msi_path, msi_opts = {}, opts = {})
 
             # (PA-514) value for PUPPET_AGENT_STARTUP_MODE should be present in
             # registry and honored after install/upgrade.
-            reg_query_command = %Q(reg query "HKLM\\SOFTWARE\\Wow6432Node\\Puppet Labs\\PuppetInstaller" /v "RememberedPuppetAgentStartupMode" | findstr #{msi_opts['PUPPET_AGENT_STARTUP_MODE']})
+            reg_key = host.is_x86_64? ? "HKLM\\SOFTWARE\\Wow6432Node\\Puppet Labs\\PuppetInstaller" :
```

Comment:
This looks good conceptually... will have to run some tests locally to make sure we're gravy.